File: examples/python/authentication/check_auth_status.py
"""
Check Authentication Status

Verifies if you're currently authenticated and displays your basic info.

Usage:
    python check_auth_status.py

Requirements:
    pip install requests
"""

import requests
import os
from datetime import datetime


def check_auth_status(cookie):
    """
    Check if authenticated and display user info.

    Args:
        cookie (str): Your _ncfa cookie value

    Returns:
        dict: Authentication status and user info
    """
    try:
        response = requests.get(
            'https://www.geoguessr.com/api/v3/profiles',
            cookies={'_ncfa': cookie}
        )

        if response.status_code == 401:
            print('❌ Not authenticated')
            print('Cookie is invalid or expired.')
            return {'authenticated': False}

        response.raise_for_status()
        profile = response.json()
        user = profile['user']

        # Display authentication status
        print(f"\n✅ Authenticated\n")
        print(f"Username: {user['nick']}")
        print(f"User ID: {user['id']}")
        print(f"Email: {profile['email']}")
        print(f"Country: {user['countryCode'].upper()}")
        print(f"Pro User: {'✅ Yes' if user['isProUser'] else '❌ No'}")
        print(f"Level: {user['progress']['level']}")
        print(f"XP: {user['progress']['xp']:,}")

        if 'competitive' in user and user['competitive']:
            print(f"\n🏆 Competitive:")
            print(f"Rating: {user['competitive']['rating']}")
            if 'division' in user['competitive'] and user['competitive']['division']:
                print(f"Division: {user['competitive']['division'].get('type', 'N/A')}")

        # Account age
        created = datetime.fromisoformat(user['created'].replace('Z', '+00:00'))
        age_in_days = (datetime.now(created.tzinfo) - created).days
        age_in_years = age_in_days / 365

        print(f"\n📅 Account Info:")
        print(f"Created: {created.strftime('%Y-%m-%d')}")
        print(f"Age: {age_in_years:.1f} years ({age_in_days} days)")

        return {
            'authenticated': True,
            'username': user['nick'],
            'userId': user['id'],
            'email': profile['email'],
            'isProUser': user['isProUser'],
            'level': user['progress']['level']
        }

    except requests.exceptions.HTTPError as e:
        print(f"❌ HTTP Error: {e}")
        return {'authenticated': False, 'error': str(e)}
    except requests.exceptions.RequestException as e:
        print(f"❌ Request Error: {e}")
        return {'authenticated': False, 'error': str(e)}


def main():
    # Get credentials from environment variables
    cookie = os.getenv('GEOGUESSR_COOKIE')
    if not cookie:
        print("❌ Error: GEOGUESSR_COOKIE environment variable not set")
        print("Set it with: export GEOGUESSR_COOKIE='your_cookie_value'")
        return

    # Check authentication
    auth = check_auth_status(cookie)

    if auth['authenticated']:
        print("\n✅ Authentication verified!")
    else:
        print("\n❌ Not authenticated")


if __name__ == '__main__':
    main()

File: examples/javascript/authentication/check-auth-status.js
/**
 * Check Authentication Status
 *
 * Verifies if you're currently authenticated and displays your basic info.
 *
 * Usage in browser console:
 * 1. Navigate to geoguessr.com
 * 2. Paste this code into the console
 * 3. Call: checkAuthStatus()
 */

async function checkAuthStatus() {
    try {
        const response = await fetch('https://www.geoguessr.com/api/v3/profiles', {
            credentials: 'include'
        });

        if (response.status === 401) {
            console.log('❌ Not authenticated');
            console.log('Please log in to GeoGuessr first.');
            return { authenticated: false };
        }

        if (!response.ok) {
            throw new Error(`HTTP ${response.status}: ${response.statusText}`);
        }

        const profile = await response.json();
        const user = profile.user;

        // Display authentication status
        console.log(`\n✅ Authenticated\n`);
        console.log(`Username: ${user.nick}`);
        console.log(`User ID: ${user.id}`);
        console.log(`Email: ${profile.email}`);
        console.log(`Country: ${user.countryCode.toUpperCase()}`);
        console.log(`Pro User: ${user.isProUser ? '✅ Yes' : '❌ No'}`);
        console.log(`Level: ${user.progress.level}`);
        console.log(`XP: ${user.progress.xp.toLocaleString()}`);

        if (user.competitive) {
            console.log(`\n🏆 Competitive:`);
            console.log(`Rating: ${user.competitive.rating}`);
            console.log(`Division: ${user.competitive.division?.type || 'N/A'}`);
        }

        // Account age
        const created = new Date(user.created);
        const ageInDays = Math.floor((Date.now() - created) / (1000 * 60 * 60 * 24));
        const ageInYears = (ageInDays / 365).toFixed(1);

        console.log(`\n📅 Account Info:`);
        console.log(`Created: ${created.toLocaleDateString()}`);
        console.log(`Age: ${ageInYears} years (${ageInDays} days)`);

        return {
            authenticated: true,
            username: user.nick,
            userId: user.id,
            email: profile.email,
            isProUser: user.isProUser,
            level: user.progress.level
        };

    } catch (error) {
        console.error('❌ Error checking authentication:', error.message);
        return { authenticated: false, error: error.message };
    }
}

// Example usage:
// checkAuthStatus();

// Use in conditional logic:
// async function doSomethingIfAuthenticated() {
//     const auth = await checkAuthStatus();
//     if (auth.authenticated) {
//         console.log('User is logged in, proceed...');
//     } else {
//         console.log('User is not logged in');
//     }
// }

// Check if Pro user:
// async function isProUser() {
//     const auth = await checkAuthStatus();
//     return auth.authenticated && auth.isProUser;
// }

// Get current user ID:
// async function getCurrentUserId() {
//     const auth = await checkAuthStatus();
//     return auth.authenticated ? auth.userId : null;
// }
